sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaignstatistics/CampaignStatisticsGrid.java

```diff
@@ -0,0 +1,99 @@
+package de.symeda.sormas.ui.campaign.campaignstatistics;
+
+import java.util.stream.Collectors;
+
+import com.vaadin.data.provider.DataProvider;
+import com.vaadin.shared.data.sort.SortDirection;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.campaign.CampaignJurisdictionLevel;
+import de.symeda.sormas.api.campaign.statistics.CampaignStatisticsCriteria;
+import de.symeda.sormas.api.campaign.statistics.CampaignStatisticsDto;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.utils.SortProperty;
+import de.symeda.sormas.ui.utils.FilteredGrid;
+
+public class CampaignStatisticsGrid extends FilteredGrid<CampaignStatisticsDto, CampaignStatisticsCriteria> {
+
+	public CampaignStatisticsGrid(CampaignStatisticsCriteria criteria) {
+		super(CampaignStatisticsDto.class);
+		setSizeFull();
+
+		setDataProvider();
+		setCriteria(criteria);
+
+		addDefaultColumns();
+		setColumnsVisibility(criteria.getGroupingLevel());
+	}
+
+	protected void addDefaultColumns() {
+		setColumns(
+			CampaignStatisticsDto.CAMPAIGN,
+			CampaignStatisticsDto.FORM,
+			CampaignStatisticsDto.AREA,
+			CampaignStatisticsDto.REGION,
+			CampaignStatisticsDto.DISTRICT,
+			CampaignStatisticsDto.COMMUNITY,
+			CampaignStatisticsDto.FORM_COUNT);
+
+		for (Column<?, ?> column : getColumns()) {
+			column.setCaption(I18nProperties.getPrefixCaption(CampaignStatisticsDto.I18N_PREFIX, column.getId(), column.getCaption()));
+		}
+	}
+
+	public void reload() {
+		getDataProvider().refreshAll();
+	}
+
+	public void setDataProvider() {
+		DataProvider<CampaignStatisticsDto, CampaignStatisticsCriteria> dataProvider = DataProvider.fromFilteringCallbacks(
```

Comment:
I think we should use an eager data provider here instead for now since lazy loading isn't implemented in the backend anyway; see any `setEagerDataProvider` method in one of the existing grids for reference